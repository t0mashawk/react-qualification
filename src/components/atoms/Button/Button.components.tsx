import React from 'react'
import * as Styled from './Button.styles'

interface ButtonProps{
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
    color?: string
    text: string
}

function Button(props: ButtonProps){
    const {onClick, color, text} = props
    return(
        <Styled.Container color={color && color}>
        <button onClick={event => onClick && onClick(event)}>{text}</button>
        </Styled.Container>
    )
}

export default Button