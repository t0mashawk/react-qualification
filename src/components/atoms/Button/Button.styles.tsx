import styled from 'styled-components'

// export default styled.button`
//     border:none;
//     background-color:red;
//     border-radius:5px;
// `

export const Container = styled.button`
 >button{
background: transparent;
border-radius: 3px;
border: 2px solid palevioletred;
color: palevioletred;
margin: 0 1em;
padding: 0.25em 1em;
 }
`