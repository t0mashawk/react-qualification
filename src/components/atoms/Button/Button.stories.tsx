import React from "react";
import { storiesOf } from "@storybook/react";
import Button from "./Button.components"

storiesOf("Button", module)
    .add("default",
        () => <Button text="Test"/>
    )
    .add("colored-button",
        () => <Button color="navy" text="Test"/>
    )