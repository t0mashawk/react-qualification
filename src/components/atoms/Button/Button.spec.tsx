import React from "react";
import renderer from "react-test-renderer"

import Button from "./Button.components"

test("Button renders as expected", () => {
  const component = renderer.create(
    <Button text="Example" color="red"/>
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});