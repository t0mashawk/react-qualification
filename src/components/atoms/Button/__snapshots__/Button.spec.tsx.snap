// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as expected 1`] = `
<div
  className="sc-AxjAm jVUdXZ"
  color="red"
>
  <button
    onClick={[Function]}
  >
    Example
  </button>
</div>
`;
